Docs: Translation memory reworked (#8290)
WeblateOrg/weblate: docs/admin/memory.rst

.. _memory-scopes:

> Translation-memory scopes

Reviewer: Nobody calls it this way.

https://en.wikipedia.org/wiki/Translation_memory, https://omegat.org/, https://www.trados.com/solutions/translation-memory/

-------------------------

.. versionadded:: 3.2

   In earlier versions translation memory could be only loaded from a file
   corresponding to the current imported translation memory scope.

> The translation memory scopes allow both privacy and sharing translations.

Author: ```suggestion
The translation memory scopes sharing translations with only relevant components, and projects, or for the entire platform.
```
Does it allow sharing between teams, users, or something else too?

Reviewer 2: This part of the article describes only that there are several scopes. Their feature are described further down, so it is not needed to list it here.

Author: It is quite difficult to understand what "translation memory scopes" are.
Something that allows for privacy, and sharing translations somehow doesn't clear that up.

.. _memory-scopes:

> Translation memory scopes

Reviewer 2: I don’t have the time or will to spend more of any on this.

   In earlier versions translation memory could be only loaded from a file
   corresponding to the current imported translation memory scope.

> The translation memory scopes ensure privacy, while enabling sharing translations.

Author: ```suggestion
The translation memory scopes allow for privacy when sharing translations.
```
Privacy is never ensured, as that is a dangerous claim.
Having any privacy isn't a product of turning on the translation memory sharing.
This makes sense to me:
`Sharing the translation memory has fine-tuned settings to adjust the scope of how and where sharing occurs.`

Reviewer 2: You are connecting here two things into one. There is a choice to keep TM private or share it. Together it won’t go well.

Reviewer 2: made the change to:

```
The translation-memory scopes ensure privacy for different projects and users. 
Sharing of translations is also available.
```

    is provided for interoperability with other tools.

.. seealso::

    :ref:`schema-memory`

.. image:: /screenshots/memory.png

.. versionadded:: 3.8

You can search for translations in the view built for this.

Admin interface
+++++++++++++++

.. versionadded:: 3.5

There is a platform-wide interface to manage the translation memory.

.. versionadded:: 4.12

It has the same options as the user-interface, but also allows
rebuilding parts of or the entire translation memory.
All old entries can be flushed and re-created from a component or project by
selecting "Administration" from amidst the different tabs
at the top of the screen, and then "Translation memory".

Management interface
++++++++++++++++++++

Several management commands can manipulate translation-memory content.
These operate on the translation memory as a whole, unfiltered by scopes
(unless requested by parameters):

:djadmin:`dump_memory`
    Exports the memory into JSON
:djadmin:`import_memory`
    Imports TMX or JSON files into the translation memory

.. versionadded:: 4.14

The Weblate API covers the translation memory.
This allows automating manipulation for different purposes,

> or based on events in the translation cycle.

Author: This part is guesswork that could possibly be improved by some realistic scenarios.
